### PR TITLE
make sure nvtx3 is used

### PR DESCRIPTION
**Pull request by jtrmal**
As far as I know, this is what nvToolsExt repo recommends (https://github.com/NVIDIA/NVTX) and they make it sound older version (sub-3) is ancient. Not sure how valid that claim is, but latest cuda docker (docker.io/nvidia/cuda:12.6.1-cudnn-devel-ubuntu22.04) needs this changes to compile -- otherwise it will end up complaining about mixing nvtx versions -- see https://github.com/jtrmal/kaldi/actions/runs/10830626409/job/30050811732#step:7:6551

I don't have much access to different grids/setups so not sure if this would be good enough or it will cause troubles -- in that  case we would have to somehow detect nvtx3 and create  define for conditional compiling. Or just get nvtx3 from github -- version 3 is header-only, so doesn't need any compilation and then linkage with a resulting library

TLDR: official docker image nvidia/cuda:12.6.1-cudnn-devel-ubuntu22.04 doesn't compile without this

### CODE REVIEW

**Comment by jtrmal** [comment]
I verified kaldi compiles with this patch even on nvidia/cuda:11.8.0-cudnn8-devel-ubuntu20.0 (generation older drivers).
So it should be fine -- feel free to chime in, but I plan to merge soon-ish (early next week) to be able to progress on the other work that is blocked by this.

**Comment by jtrmal** [comment]
I'll merge this and we will see what happens :)

**Comment by danpovey** [comment]
we got that same failure with lattice-utils-test.

**Comment by jtrmal** [comment]
yeah, that wasn't supposed to fix that -- that was fixing some compilation
issues in ubuntu docker
The log of the failed test is here:
https://github.com/kaldi-asr/kaldi/actions/runs/10886905072/artifacts/1938314600
y.

**Comment by danpovey** [comment]
It looks like it might possibly be a failure with OpenFST's ShortestPath algorithm with n > 1 (i.e. requesting more than one path in n-best).  Does anyone have time to see whether they made any changes to that algorithm from 1.7.x to 1.8.x?

**Comment by danpovey** [comment]
I tried to have a look at diffs of the OpenFst code but don't see anything that looks very relevant.  It will probably be necessary to see the FSTs in the failing test.

**Comment by jtrmal** [comment]
Last time I was looking at this, the fats looked the same only the weights
were different (it was as if the second part of the composite lattice
weight disappeared, ie was eg 15 instead of 15, 0_1)
Y.
